### ISSUE TITLE
RedundantBraces fails on String interpolation with backquoted variable name

### ISSUE THREAD

**Issue by gaelrenoux** [scalameta/scalafmt#1227]
- **Version**: 1.6.0-RC3
- **Configuration**:
```
rewrite {
  rules = [ RedundantBraces ]
  redundantBraces.stringInterpolation = true
}
```

## Steps

This is the code to format:
```scala
class Thing(`type`: String) {
  def label = s"type=${`type`}"
}
```

Note that the bug has nothing to do with `type` being a keyword (any backquoted name fails), but having a variable named as a keyword is usually the reason why you put backquotes.

I run scalafmt using the CLI or SBT (no option):

```bash
scalafmt
sbt scalafmt
```

## Problem

Scalafmt fails:
```
/Users/gael/Repositories/Local/tests/src/main/scala/Thing.scala:2: error: `$$', `$'ident, `$'this, `$'BlockExpr`$'_
  def label = s"type=$`type`"
```

It looks like scalafmt tries to remove the braces.

## Expectation

It shouldn't fail, and it should leave the braces in place (they are necessary).

## Workaround

Using an intermediate value solves the problem:
```scala
class Thing(`type`: String) {
  private val typ = `type`
  def label = s"type=${typ}"
}
```

When I run scalafmt on this code, the curly braces inside the interpolation are removed (as expected).

**Comment by olafurpg** [comment]
Thanks for reporting! I am able to reproduce, this is a bug. The rewrite rule should not be triggered for backticked identifiers.

**Comment by poslegm** [comment]
@gaelrenoux please try it with scalafmt [2.0.0-RC8](https://github.com/scalameta/scalafmt/releases/tag/v2.0.0-RC8)

**Comment by poslegm** [comment]
Fixed in 2.0.0-RC8
